data/transactions/logic/eval.go

```diff
@@ -824,8 +812,15 @@ func EvalSignature(gi int, params *EvalParams) (pass bool, err error) {
 func eval(program []byte, cx *EvalContext) (pass bool, err error) {
 	defer func() {
 		if x := recover(); x != nil {
-			err = makePanicError(x, cx.EvalParams, "Eval")
+			buf := make([]byte, 16*1024)
+			stlen := runtime.Stack(buf, false)
 			pass = false
+			errstr := string(buf[:stlen])
+			if cx.Trace != nil {
```

Comment:
Is there ever a case where cx itself could be nil?

Reply:
No, `eval` is invoked only from `EvalSignatureFull` and `EvalContract`, which are guaranteed to pass in non-nil values for `cx`